qlib/rl/aux_info.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from __future__ import annotations
+
+from typing import Generic, TYPE_CHECKING, TypeVar
+
+from qlib.typehint import final
+
+from .simulator import StateType
+
+if TYPE_CHECKING:
+    from .utils.env_wrapper import EnvWrapper
+
+
+__all__ = ["AuxiliaryInfoCollector"]
+
+AuxInfoType = TypeVar("AuxInfoType")
+
+
+class AuxiliaryInfoCollector(Generic[StateType, AuxInfoType]):
+    """Override this class to collect customized auxiliary information from environment."""
+
+    env: "EnvWrapper" | None = None
```

Comment:
No we can't. Otherwise we will have circular imports.

Reply:
I think it will not trigger circular imports because `from .utils.env_wrapper import EnvWrapper` is wrapped by `if TYPE_CHECKING:`. The import operation will not be executed in runtime.